Added spectrum testing scripts

Cirromulus opened the pull request:
Added scripts for analyzing your neighboring frequencies for interference hunting.

Cirromulus committed:
Added spectrum testing scripts

Cirromulus committed:
Merge branch 'master' into add-waterfall-testscripts

Cirromulus committed:
added image

Cirromulus commented:
About the image: Leave it embedded (big image, kind of disturbs reading flow) or just link to it?

jekhokie commented on docs/troubleshooting.md:
```diff
@@ -53,12 +53,31 @@ The frequency used for the `test_reception.sh` script should be heard and can be
 
 Replace 97.3 with a strong local FM station. If you have an FM trap or LNA you may need to remove them from the antenna feed.
 
+## Noise level and interference sources
```
Change to a single-level heading `#` for consistency.

jekhokie commented on docs/troubleshooting.md:
````diff
@@ -53,12 +53,31 @@ The frequency used for the `test_reception.sh` script should be heard and can be
+```
+This will scan a range for five hours and produce a heatmap waterfall image afterwards.
+Alternatively, you can call `start_scanning.sh` manually.
+If required (or curious), you can change the freqency to scan there, too.
````
You might add where the output file that is generated would end up going so people know exactly where to find it when it completes.

jekhokie commented on scripts/testing/.gitignore:
```diff
@@ -0,0 +1,4 @@
+nohup.out
```
Let's move this nohup out of this directory please and into the `tmp/` directory in the raspberry-noaa-v2 folder. I think this is a better move so we don't get too crazy about temp files that don't matter to the framework being littered all over the place.

jekhokie commented on scripts/testing/.gitignore:
```diff
@@ -0,0 +1,4 @@
+nohup.out
+Vera.ttf
```
This is needed to generate the graphs, correct? If so, shouldn't this be committed to the source repo?

Cirromulus replied:
It is not needed, this is automatically downloaded once by `heatmap.py`.
I left this out, in case some day `heatmap.py` does something else or in a system where the font is already installed.

jekhokie commented on scripts/testing/.gitignore:
```diff
@@ -0,0 +1,4 @@
+nohup.out
+Vera.ttf
+*.csv.gz.*
```
Same comment here about where to store things - let's update this so that the output files end up in `tmp/spectrum`, and create that directory using Ansible (see `ansible/roles/common/tasks/tools.py` for how the other temp directories are created).

Cirromulus replied:
I now suggest working from home dir to prevent an unclean repo state.
Also, it deletes the intermediate files after generating output.

If you want it in the website, we could copy/move the resulting file somewhere meaningful.

jekhokie commented on scripts/testing/generate_waterfall.sh:
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
```
Please add a purpose and example usage comment to the top of these scripts for consistency.

jekhokie commented on scripts/testing/heatmap.py:
```diff
@@ -0,0 +1,625 @@
+#! /usr/bin/env python
```
Please update the header here as a comment that indicates where you retrieved this script (to give credit).

jekhokie commented on scripts/testing/scan_for.sh:
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
```
Add comment for purpose/inputs/example usage.

jekhokie commented on scripts/testing/start_scanning.sh:
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
```
Add comment for purpose/inputs/example usage.

jekhokie commented on scripts/testing/stop_and_finalize_scanning.sh:
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```
Add comment for purpose/inputs/example usage.

jekhokie commented on docs/troubleshooting.md:
```diff
@@ -53,12 +53,31 @@ The frequency used for the `test_reception.sh` script should be heard and can be
+If required (or curious), you can change the freqency to scan there, too.
+
+Example analysis:
+![waterfall analysis](./assets/images/scan_annotated.jpg)
```
I agree with your question - let's just paste a link here (instead of embedding the image) so when the user clicks on the link, it brings up the actual image from where it's located currently.

jekhokie reviewed (changes requested):
@Cirromulus this is awesome thanks for submitting it. I left comments on everything, and one major thing I'd like to see changed is to use the `common.sh` script and NOAA vars such that you can output the temporary and/or generated files (csv, gzip, etc.) into a `${NOAA_HOME}/tmp/spectrum/` directory, which should be created by Ansible.

Also, please add a comment in the README.md under `Credits` indicating where you obtained the scripts and capabilities in this PR to give credit where credit is due :)

Cirromulus committed:
polished docs

Cirromulus committed:
Removed the need for storing output or intermediate *csv.gz. Now suggesting working from home dir to prevent clatter.

Cirromulus commented:
> Also, please add a comment in the README.md under `Credits` indicating where you obtained the scripts and capabilities in this PR to give credit where credit is due :)

We can do this. This is the inital author of some important features of our rtl-sdr drivers.

Cirromulus committed:
Mention kmkeen

jekhokie reviewed (commented):
@Cirromulus I just noticed that the `docs/assets/images/scan_annotated.jpg` file included in this PR is showing an "Empty file." I'm assuming this is not expected?

jekhokie committed:
Small Tweaks to Scanner Scripts

Some small updates to the scanner scripts and adding the font
file/removing the call-out to the font file which removes the need for
remote connectivity for running the scanner scripts.

jekhokie commented:
@Cirromulus nice work - I made a few tweaks to button things up, but merging now!

Cirromulus commented:
> @Cirromulus I just noticed that the `docs/assets/images/scan_annotated.jpg` file included in this PR is showing an "Empty file." I'm assuming this is not expected?

It works if you click on "rich diff". I don't know whether this is normal or github has a hiccup.
![image](https://user-images.githubusercontent.com/7480344/109278308-ac49d480-7818-11eb-8dd4-06915297a746.png)
